web/src/integration_url_modal.js

```diff
@@ -37,7 +38,26 @@ export function show_generate_integration_url_modal(api_key) {
         const $topic_input = $("#integration-url-topic-input");
         const $integration_url = $("#generate-integration-url-modal .integration-url");
         const $dialog_submit_button = $("#generate-integration-url-modal .dialog_submit_button");
+        const $show_integration_events = $("#show-integration-events");
 
+        $show_integration_events.on("change", () => {
+            $("#integration-events-ui").toggleClass(
+                "hide",
+                !$show_integration_events.prop("checked"),
+            );
+            update_url();
+        });
+        $(document).on("change", ".integration-event-checkbox", () => {
+            update_url(true);
+        });
+        $("#check-all-events").on("click", () => {
+            $(".integration-event-checkbox").prop("checked", true);
+            update_url(true);
+        });
+        $("#uncheck-all-events").on("click", () => {
+            $(".integration-event-checkbox").prop("checked", false);
+            update_url(true);
+        });
```

Comment:
I might move these listeners to be after the $override_topic listener, again for readability.